app/scripts/metamask-controller.js

```diff
@@ -7821,18 +7828,25 @@ export default class MetamaskController extends EventEmitter {
       }),
     );
 
-    engine.push(filterMiddleware);
-    engine.push(subscriptionManager.middleware);
-
-    engine.push(this.metamaskMiddleware);
-
     engine.push(this.eip5792Middleware);
 
     if (subjectType === SubjectType.Snap && isSnapPreinstalled(origin)) {
       engine.push(this.eip7702Middleware);
     }
 
-    engine.push(providerAsMiddleware(proxyClient.provider));
+    engine.push(filterMiddleware);
```

Comment:
I recall that re-ordering these two middleware functions has caused problems in the past 🤔 

Maybe fine in this case, I will take a closer look.

Reply:
Could we introduce another JsonRpcEngineV2 wrapper instead, to preserve the order?

e.g.

```
    engine.push(
      asLegacyMiddleware(
        JsonRpcEngineV2.create({
          middleware: this.metamaskMiddlewareV2,
        }),
      ),
    );
```